torch/fx/symbolic_trace.py

```diff
@@ -178,6 +183,34 @@ def search_for_tensor(m : torch.nn.Module) -> Optional[List[str]]:
 def _proxy_placeholder(name: str, delegate: DelegateBase) -> Proxy:
     return Proxy(delegate.placeholder(name), delegate)
 
+# Context manager to specify--for all symbolically traced Nodes within the
+# `with` block, the qualified name of the Module from which those nodes
+# originated. Note that multiple ModuleHierarchyCtxMgr's can be nested
+# reentrantly.
+class ModuleHierarchyCtxMgr:
+    def __init__(self, target : str):
+        self.target = target
+
+    def __enter__(self):
+        if not torch.jit.is_scripting():
```

Comment:
Explicit code to make this scriptable breaks the composability of these transforms. It's much better to have the code be naturally scriptable.